lib/driver.js

```diff
@@ -410,460 +391,399 @@ DatastaxDriver.prototype.executeCqlOnDriver = function executeCqlOnDriver(pool,
+          }
+          debugParams.push(Buffer.isBuffer(p) ? '<buffer>' : JSON.stringify(p));
+        });
+      }
+      this.logger.debug('priam.cql: executing cql statement', {
+        consistencyLevel: options.consistency,
+        cql: cqlQuery,
+        params: debugParams
+      });
+    }
   }
-  return executeRetryableCql();
 
-};
-
-DatastaxDriver.prototype.prepareQueryArgs = function prepareQueryArgs(pool, cqlStatement, params, consistency, options) {
-  var self = this;
-  var execOptions = _.assign({
-    prepare: !!options.executeAsPrepared,
-    consistency: consistency
-  }, options);
-
-  var hints = [];
-  var routingIndexes = [];
-  _.forEach(params, function (param, index) {
-    if (param && param.hasOwnProperty('value') && param.hasOwnProperty('hint')) {
-      params[index] = param.value;
-      if (param.hint) {
-        if (_.isString(param.hint)) {
-          param.hint = self.dataType.getByName(param.hint);
-        }
-        hints[index] = param.hint;
+  async _timeQuery(queryRequestId, cqlQuery, dataParams, options, act) {
+    try {
+      const metrics = this.metrics;
+      const captureMetrics = !!(metrics && options.queryName);
+      const start = captureMetrics && process.hrtime();
+      this.emit('queryStarted', queryRequestId, cqlQuery, dataParams, options.consistency, options);
+      const result = await act(options);
+      this.emit('queryCompleted', queryRequestId);
+      if (captureMetrics) {
+        var duration = process.hrtime(start);
+        duration = (duration[0] * 1e3) + (duration[1] / 1e6);
+        metrics.measurement('query.' + options.queryName, duration, 'ms');
       }
-      if (param.isRoutingKey) {
-        routingIndexes.push(index);
+      return result;
+    } catch (err) {
+      this.emit('queryFailed', queryRequestId);
+      throw err;
+    }
+  }
+
+  _prepareQueryArgs(pool, cqlStatement, params, consistency, options) {
+    var self = this;
+    var execOptions = _.assign({
+      prepare: !!options.executeAsPrepared,
+      consistency: consistency
+    }, options);
+
+    var hints = [];
+    var routingIndexes = [];
+    _.forEach(params, function (param, index) {
+      if (param && param.hasOwnProperty('value') && param.hasOwnProperty('hint')) {
+        params[index] = param.value;
+        if (param.hint) {
+          if (_.isString(param.hint)) {
+            param.hint = self.dataType.getByName(param.hint);
+          }
+          hints[index] = param.hint;
+        }
+        if (param.isRoutingKey) {
+          routingIndexes.push(index);
+        }
       }
+    });
+    if (hints.length) {
+      execOptions.hints = hints;
     }
-  });
-  if (hints.length) {
-    execOptions.hints = hints;
-  }
-  if (routingIndexes.length) {
-    execOptions.routingIndexes = routingIndexes;
+    if (routingIndexes.length) {
+      execOptions.routingIndexes = routingIndexes;
+    }
+    return { cqlStatement: cqlStatement, params: params, execOptions: execOptions };
```

Comment:
Can collapse this styling since the param names match the variables

Reply:
@scommisso: done